File: solutions/_523_continuous_subarray_sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var checkSubarraySum = function (nums, k) {
  let sum = 0;
  let map = new Map();
  map.set(0, -1);

  for (let i = 0; i < nums.length; i++) {
    sum += nums[i];
    if (k !== 0) {
      sum %= k;
    }

    if (map.has(sum)) {
      if (i - map.get(sum) > 1) {
        return true;
      }
    } else {
      map.set(sum, i);
    }
  }

  return false;
};

module.exports = { checkSubarraySum };

/*
https://leetcode.com/problems/continuous-subarray-sum/
*/

File: solutions/_523_continuous_subarray_sum/solution.py
from typing import List


class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        map = {}
        map[0] = -1
        sum = 0
        for i, num in enumerate(nums):
            sum += num
            sum %= k
            if sum in map:
                if i - map[sum] > 1:
                    return True
            else:
                map[sum] = i
        return False


# https://leetcode.com/problems/continuous-subarray-sum/
